### PR TITLE
Add CB CI tests

### PR DESCRIPTION
**Pull request by as-suvorov**

### CODE REVIEW

File: .github/workflows/continuous_batching_cpp.yml
Comment (Wovchena) [inline .github/workflows/continuous_batching_cpp.yml:1]: I noticed that the author receives e as many mails as there are yml files when everything fails. To reduce the spam, please move sample tests to causal_lm_cpp.yml and python tests to genai_python_lib.yml.

Comment (as-suvorov) [reply .github/workflows/continuous_batching_cpp.yml:1]: Would disabling workflow notifications work for you: https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#github-actions-notification-options ?

Comment (Wovchena) [reply .github/workflows/continuous_batching_cpp.yml:1]: No. The notifications are useful. But it's enough to send it only once per commit.

Comment (as-suvorov) [reply .github/workflows/continuous_batching_cpp.yml:1]: Ok, done

**Review by Wovchena — APPROVED** [review]

File: tests/python_tests/test_sampling.py [hunk 1]
```diff
@@ -148,6 +156,7 @@ class RandomSamplingTestStruct:
                                     '\n\nOpenVINO is a social networking tool. OpenVINO is a free virtualization service that works at scale. The tool provides the ability'
                                 ]
                             ]),
+                            marks=[pytest.mark.xfail(reason="assert ref_text == ov_text fails in CI.", strict=True, condition=sys.platform in ["darwin", "win32"])]),
```
Comment (ilya-lavrenov) [inline tests/python_tests/test_sampling.py:159]: is it an issue because of different implementation of random in C++ on Ubuntu and Windows?

Comment (ilya-lavrenov) [reply tests/python_tests/test_sampling.py:159]: Ok, as I see it's implemented in this PR https://github.com/openvinotoolkit/openvino.genai/pull/628

**Pull request merged**